### PR TITLE
Show nightly version

### PR DESCRIPTION
**Pull request by allewun**
@kevgo @charlierudolph 

What do you think... useless or not? I frequently wonder exactly what revision I'm using, esp as it's been a while since 0.5.0.. though I tend follow it up with a `git sync` in `git-town/` anyway.

Maybe useful for those that installed via the manual instructions?

### CODE REVIEW

**Comment by kevgo** [comment]
We don't have nightly builds. What's the use case for this? 

Also, can you please provide an example output?

**Comment by allewun** [comment]
"Nightly" might be the wrong term. For people who've installed git town manually, the version they use likely has a different feature set than what the release notes indicates, even though `git town version` would print 0.5.0.

**Comment by allewun** [comment]
And just to clarify, if Git Town is installed via homebrew, the output remains unchanged:
```
> git town version
Git Town 0.5.0
```

**Comment by charlierudolph** [comment]
So all you want is the date and commit sha to show in the version?
That seems fine to me. I would prefer a more readable version
```
> git town version
Git Town 0.5.0 (Mar 31, 2015 - b7fe18d) 
```
I think adding the date and sha always would be fine too.

Along that line we need to release more. I'll start preparing a release now actually

**Comment by allewun** [comment]
Updated:

**Comment by kevgo** [comment]
I'm still somehow YAGNI on this. I'd like to caution us to keep Git Town as simple as possible, and only add things for which there is a real user demand, and which solve a real problem. If that's the case for you @allewun then I'm fine merging this. But if not, I'd rather leave it out. Up to you.

**Comment by allewun** [comment]
Completely agree about keeping things simple. This is a fairly low-impact change though. If it causes any issues later on, I have no issue with ripping it out. For the moment, I see it being marginally useful, so let's give it a shot.

File: src/helpers/environment.sh [hunk 1]
```diff
@@ -16,6 +17,17 @@ function ensure_git_repository {
 }
 
 
+function gittown_dev_version {
+  local base_dir=$( dirname "${BASH_SOURCE[0]}" )
```
Comment (charlierudolph) [inline src/helpers/environment.sh]: remove space padding inside parens

File: src/helpers/environment.sh [hunk 2]
```diff
@@ -16,6 +17,17 @@ function ensure_git_repository {
...
+  local git_hash=$(git -C "$base_dir" rev-parse --short HEAD)
+  local git_date=$(git -C "$base_dir" --no-pager show -s --format=%cD HEAD | cut -d ' ' -f2-4)
+
+  if ! git -C "$base_dir" remote -v > /dev/null 2>&1 | grep "https://github.com/Homebrew/homebrew.git (fetch)"; then
```
Comment (charlierudolph) [inline src/helpers/environment.sh]: Why the output redirection? And shoudln't this whole thing be in a subshell?

File: src/helpers/environment.sh [hunk 3]
```diff
@@ -16,6 +17,17 @@ function ensure_git_repository {
 }
 
 
+function gittown_dev_version {
+  local base_dir=$( dirname "${BASH_SOURCE[0]}" )
+  local git_hash=$(git -C "$base_dir" rev-parse --short HEAD)
+  local git_date=$(git -C "$base_dir" --no-pager show -s --format=%cD HEAD | cut -d ' ' -f2-4)
```
Comment (charlierudolph) [inline src/helpers/environment.sh]: Dont calculate these unless you need them (move then to the inside of the if statement.

File: src/helpers/environment.sh (hunk 2 above)
Comment (allewun) [inline src/helpers/environment.sh]: Why should it be in a subshell? The output redirection is in case the user isn't using Git 1.8.5 and above and the [`-C`](https://github.com/git/git/blob/5fd09df3937f54c5cfda4f1087f5d99433cce527/Documentation/RelNotes/1.8.5.txt#L115-L116) flag isn't supported.

Related to that, have we been considering the version of Git that users will have?

Comment (charlierudolph) [inline src/helpers/environment.sh]: All calls where we want to capture the output should be in a subshell.As then the output is returned and we can compare it

Comment (allewun) [inline src/helpers/environment.sh]: I'm not capturing the output here, I'm only seeing if grep returned successfully.

* success => git-town was installed via homebrew
* failure => git-town was installed some other way

Comment (charlierudolph) [inline src/helpers/environment.sh]: I suggest against that. Instead you can have grep output the count of matches. This is what we have been doing throughout the repository. 
```sh
if [ "$(git -C "$base_dir" remote -v | grep -c "https://github.com/Homebrew/homebrew.git (fetch)")" == 0 ]; then
  # ...
fi
```

Comment (allewun) [inline src/helpers/environment.sh]: Alrighty. TBH I don't see any benefit doing that, but for consistency I'll change it

Comment (allewun) [inline src/helpers/environment.sh]: I guess the negation might be hard to follow

Comment (charlierudolph) [inline src/helpers/environment.sh]: I disliked the negation and the output redirection.

File: src/helpers/environment.sh [hunk 4]
```diff
@@ -16,6 +17,18 @@ function ensure_git_repository {
 }
 
 
+function gittown_dev_version {
+  local base_dir=$(dirname "${BASH_SOURCE[0]}")
+
+  if ! git -C "$base_dir" remote -v > /dev/null 2>&1 | grep "https://github.com/Homebrew/homebrew.git (fetch)"; then
+    local git_hash=$(git -C "$base_dir" rev-parse --short HEAD)
+    local git_date=$(git -C "$base_dir" --no-pager show -s --format=%cD HEAD | cut -d ' ' -f2-4)
```
Comment (charlierudolph) [inline src/helpers/environment.sh]: `--no-pager` should be unnecessary as the output is a single line.

Comment (charlierudolph) [inline src/helpers/environment.sh]: I'd also prefer a space between `-f` and `2-4`

Comment (allewun) [inline src/helpers/environment.sh]: done

File: src/helpers/environment.sh [hunk 5]
```diff
@@ -16,6 +17,19 @@ function ensure_git_repository {
 }
 
 
+function gittown_dev_version {
+  local base_dir=$(dirname "${BASH_SOURCE[0]}")
+
```
Comment (charlierudolph) [inline src/helpers/environment.sh]: remove one of these empty lines

File: src/git-town [hunk 6]
```diff
@@ -31,7 +31,7 @@ function show_help {
 
 
 function show_version {
-  echo "Git Town 0.5.0"
+  echo "Git Town 0.5.0$(gittown_dev_version)"
```
Comment (charlierudolph) [inline src/git-town]: This looks like dev_version is always printed when thats not the case. How about
```sh
local version="0.5.0"
if [ "$(is_manual_install)" == "true" ]; then
  version="$version ($revision_info)"
fi
echo "Git Town $version"
```

File: src/helpers/environment.sh (hunk 5 above)
Comment (allewun) [inline src/helpers/environment.sh]: which one? :D

**Comment by kevgo** [comment]
Alright, this looks good from my end. :ship:

**Comment by charlierudolph** [comment]
:ship:
